viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -19,6 +19,44 @@
 #include "cartographer/mapping/map_builder.h"
 #include "glog/logging.h"
 
+namespace {
+// The default value Cairo when coloring the image if no data is present. A
+// pixel representing no data will have all 3 channels of its color channels
+// (RGB) as the default value([102,102,102])
+static const unsigned char defaultCairosEmptyPaintedSlice = 102;
+// Number of bytes in a pixel
+const int bytesPerPixel = 4;
+struct pixelColorARGB {
+    unsigned char A;
+    unsigned char R;
+    unsigned char G;
+    unsigned char B;
+};
+
+// Check if pixel is the default color signalling no data is present
+bool checkIfEmptyPixel(pixelColorARGB color) {
+    return (color.R == defaultCairosEmptyPaintedSlice) &&
+           (color.G == defaultCairosEmptyPaintedSlice) &&
+           (color.B == defaultCairosEmptyPaintedSlice);
+}
```

Comment:
[nit]
```suggestion
bool CheckIfEmptyPixel(ColorARGB pixel_color) {
    return (pixel_color.R == defaultCairosEmptyPaintedSlice) &&
           (pixel_color.G == defaultCairosEmptyPaintedSlice) &&
           (pixel_color.B == defaultCairosEmptyPaintedSlice);
}
```

Reply:
moving all variables non consts to snake case due to google C++ style guidelines